src/wallet.ts

```diff
@@ -0,0 +1,218 @@
+import ethTx = require('ethereumjs-tx');
+import ethWallet = require('ethereumjs-wallet');
+
+declare const Buffer;
+declare const setTimeout;
+
+export default class Wallet {
+  length: number;
+  nonce: number;
+  web3: any;
+
+  constructor(web3: any) {
+    this.length = 0;
+    this.nonce = 0;
+    this.web3 = web3;
+  }
+
+  _findSafeIndex(pointer = 0) {
+    pointer = pointer;
+    if (this.hasOwnProperty(pointer)) {
+      return this._findSafeIndex(pointer + 1);
+    } else {
+      return pointer;
+    }
+  }
+
+  _currentIndexes() {
+    const keys = Object.keys(this);
+    const indexes = keys
+      .map((key) => parseInt(key, 10))
+      .filter((n) => n < 9e20)
+      .slice(0, this.length);
+    return indexes;
+  }
+
+  create(numAccounts) {
+    for (let i = 0; i < numAccounts; i++) {
+      const wallet = ethWallet.generate();
+      this.add(wallet);
+    }
+    return this;
+  }
+
+  add(wallet) {
+    if (!this[wallet.getAddressString()]) {
+      const idx = this._findSafeIndex();
+      wallet.index = idx;
+
+      this[idx] = wallet;
+      this[wallet.getAddressString()] = wallet;
+      this[wallet.getAddressString().toLowerCase()] = wallet;
+      this.length++;
```

Comment:
Is there any reason for setting so many variables to equate `wallet`?

Reply:
it just means we have 3 different fields to access the child wallet from, we might have different situations when we need to use them